lib/Parser/Parse.cpp

```diff
@@ -13904,6 +13904,7 @@ void Parser::ReleaseTemporaryGuestArena()
         if (this->m_scriptContext != nullptr)
         {
             this->m_scriptContext->ReleaseTemporaryGuestAllocator(m_tempGuestArena);
+            m_tempGuestArena.Unroot();
```

Comment:
Should the `Unroot()` call be part of ReleaseTemporaryGuestAllocator?

Reply:
No, the m_tempGuestArena is a AutoRecyclerRootPtr and hence we need to call unroot on it here.

---
In reply to: [206287410](https://github.com/Microsoft/ChakraCore/pull/5554#discussion_r206287410) [](ancestors = 206287410)